common/environment/environment.cpp

```diff
@@ -1620,11 +1612,57 @@ IConstDropZoneInfo * CLocalEnvironment::getDropZoneByAddressPath(const char * ne
                 dropzonePathLen = strlen(pdropzoneDir);
                 dropzonePath.set(pdropzoneDir);
                 pdropzonePath = dropzonePath.str();
-                retVal = &zoneIt->query();
+                dropZone = &zoneIt->query();
+            }
+        }
+    }
+    if (dropZone)
+    {
+        // check if it is the local machine
+        Owned<IConstMachineInfo> machineInfo = getMachineByAddress(netaddress);
+        if (machineInfo)
```

Comment:
I don't think this section is right.. (but I may misunderstand)

As I understand it, dropzones will be defined like this in future:
```
  <DropZone build="_"
            buildSet="DropZone"
            computer=""
            description="DropZone process"
            directory="/var/lib/HPCCSystems/mydropzone"
            ECLWatchVisible="true"
            name="dropzone_2"
            umask="022">
   <WhiteList name="WhiteList" server="2.1.1.1"/>
   <WhiteList name="WhiteList_2" server="2.4.4.4"/>
  </DropZone>
```

i.e. WhiteList - but no defined @computer name (shouldn't be there at all in the definition I'd have thought).

DropZone's with @computer are therefore legacy definitions - which we have to continue to support for backward compatibility reasons only.

So this section should be dealing with the legacy case afaics, i.e. checking if found dropzone has a @computer and if it's netAddress matches.